progress/react/demo/.gitignore

```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
+# dependencies
+/node_modules
+/.pnp
+.pnp.js
+
```

Comment:
add package-lock.json in here so it doesn't accidentally get checked in if someone uses npm

Reply:
Added this :)